etp-backend/src/main/clj/solita/etp/service/laatija.clj

```diff
@@ -10,17 +11,30 @@
 ; *** Conversions from database data types ***
 (def coerce-laatija (coerce/coercer laatija-schema/Laatija json/json-coercions))
 
-(defn add-laatija! [db laatija]
-  (:id (laatija-db/insert-laatija<! db {:data (json/write-value-as-string laatija)})))
-
 (defn find-laatija [db id]
   (first (map (comp coerce-laatija json/merge-data) (laatija-db/select-laatija db {:id id}))))
 
+(defn find-laatija-with-henkilotunnus [db henkilotunnus]
+  (first (map (comp coerce-laatija json/merge-data) (laatija-db/select-laatija-with-henkilotunnus db {:henkilotunnus henkilotunnus}))))
+
+(defn add-or-update-existing-laatijat! [ctx laatijat]
```

Comment:
ctx should be probably named db since ctx would mean the whole request context instead of just datasource.